src/geom/SpanSet.cc

```diff
@@ -0,0 +1,640 @@
+                     std::swap(goodRuns, candidateRuns);
+                 } else {
+                     std::list<PrimaryRun> newlist;
+                     for (auto & good : goodRuns) {
+                         for (auto & cand : candidateRuns) {
+                             int start = std::max(good.xmin, cand.xmin);
+                             int end = std::min(good.xmax, cand.xmax);
+                             if (end >= start) {
+                                 newlist.push_back(PrimaryRun({m, y, start, end}));
+                             }
+                         }
+                     }
+                     std::swap(newlist, goodRuns);
+                 }
+             }
+         }
+         for (auto & run : goodRuns) {
+             tempVec.push_back(geom::Span(run.y, run.xmin, run.xmax));
+         }
+    }
+    return geom::SpanSet(std::move(tempVec));
+}
+
+bool geom::SpanSet::operator==(SpanSet const & other) const {
+    // Check the equivelance of this SpanSet with another
+    return spanVector == other.spanVector;
+}
+
+bool geom::SpanSet::operator!=(SpanSet const & other) const {
+    // Check the equivelance of this SpanSet with another
+    return spanVector != other.spanVector;
+}
+
+geom::SpanSet geom::SpanSet::spanSetFromShape(int r, Stencil s) {
+    // Create a SpanSet from a given Stencil
+    std::vector<Span> tempVec;
+    tempVec.reserve(2*r + 1);
+    switch (s) {
+        case Stencil::CIRCLE:
+            for (auto dy = -r; dy <= r; ++dy) {
+                int dx = static_cast<int>(sqrt(r*r - dy*dy));
+                tempVec.push_back(geom::Span(dy, -dx, dx));
+            }
+            break;
+        case Stencil::MANHATTAN:
+            for (auto dy = -r; dy <= r; ++dy) {
+                int dx = r - abs(dy);
+                tempVec.push_back(geom::Span(dy, -dx, dx));
+            }
+            break;
+        case Stencil::BOX:
+            for (auto dy = -r; dy <= r; ++dy) {
+                int dx = r;
+                tempVec.push_back(geom::Span(dy, -dx, dx));
+            }
+            break;
+    }
+    return geom::SpanSet(std::move(tempVec), false);
+}
+
+geom::SpanSet geom::SpanSet::intersect(geom::SpanSet const & other) const {
+    // Check if the bounding boxes overlap, if not return null spanset
+    if (!bBox.overlaps(other.getBBox())) {
+        return geom::SpanSet();
+    }
+    std::vector<Span> tempVec;
+    for (auto const & spn : spanVector) {
+        for (auto const & otherSpn : other) {
+            if (spansOverlap(spn, otherSpn)) {
+                auto newMin = std::max(spn.getMinX(), otherSpn.getMinX());
+                auto newMax = std::min(spn.getMaxX(), otherSpn.getMaxX());
+                auto newSpan = geom::Span(spn.getY(), newMin, newMax);
+                tempVec.push_back(newSpan);
+            }
+        }
+    }
+    return geom::SpanSet(std::move(tempVec), false);
```

Comment:
Do you need to nomalize the SpanSet here in case one span from other overlaps two spans?